lib/ansible/playbook/included_file.py

```diff
@@ -46,7 +46,7 @@ def add_host(self, host):
             self._hosts.append(host)
 
     def __eq__(self, other):
-        return other._filename == self._filename and other._args == self._args
+        return other._filename == self._filename and other._args == self._args and other._task._parent._uuid == self._task._parent._uuid
```

Comment:
I think (but haven't tested) that if self._task is a TaskInclude, then if used in conjunction with https://github.com/ansible/ansible/pull/37304, this could be:

``` python
def __eq__(self, other):
    return other._filename == self._filename and other._args == self._args and other._task._parent == self._task._parent
```

Reply:
Possibly even 
``` python
def __eq__(self, other):
    return other._filename == self._filename and other._args == self._args and other._task == self._task
```

But, untested...